presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -429,6 +435,15 @@ public Page getOutput()
             return null;
         }
 
+        // process unfinished work if one exists
+        if (unfinishedWork != null) {
+            if (unfinishedWork.process()) {
+                unfinishedWork = null;
+            }
+            aggregationBuilder.updateMemory();
+            return null;
```

Comment:
I understand that `return null` unconditionally is ok from a functionality perspective. But this raises confusion for readers of the code.

```
            boolean workDone = unfinishedWork.process();
            aggregationBuilder.updateMemory();
            if (!workDone) {
                return null;
            }
            unfinishedWork = null;
```